### ISSUE TITLE
Go Routine Leak

### ISSUE THREAD

**Issue by ankur-a** [sanketplus/go-mysql-lock#5]
There is currently a  go-routine leak when spawning a refresher goroutine `go lock.refresher(l.refreshInterval, cancelFunc)` due to the use of an unbuffered channel `unlocker:        make(chan struct{})` , as in `l.Release()` called after connection ping fails due to cancellation of connection context, as there is no consumer listening to the channel `l.unlocker <- struct{}{}` while calling release goes to an infinite deadlock wait.

Using `unlocker:        make(chan struct{}, 1)` should fix this deadlock.

**Comment by sanketplus** [comment]
Thanks @ankur-a for digging this out. Can you raise a PR, possibly with a test covering this scenario?

**Comment by ankur-a** [comment]
Sure, will raise a PR to fix this, using `https://github.com/uber-go/goleak` this in the unit test `TestMysqlLocker_LockContext_Success` or any other method will throw errors regarding this. 
eg : 

```
func TestMysqlLocker_LockContext_Success(t *testing.T) {
	defer goleak.VerifyNone(t)
	ctx := context.Background()
	db := setupDB(t)
	key := "foo"
	lock := getLockContext(ctx, t, key, db)
	lockContext := lock.GetContext()
	releaseLock(t, lock)

	// making sure lock's context is done after lock is released
	select {
	case <-lockContext.Done():
	default:
		assert.Fail(t, "lock's context is not cancelled after lock is released")
	}
}
```

Output:
``` /Users/ankur/workspaces/go-mysql-lock/leaks.go:78: found unexpected goroutines:
        [Goroutine 20 in state select, with database/sql.(*DB).connectionOpener on top of the stack:
        goroutine 20 [select]:
        database/sql.(*DB).connectionOpener(0xc00011c000, 0x13eb440, 0xc0000e49c0)
        	/usr/local/go/src/database/sql/sql.go:1071 +0xe8
        created by database/sql.OpenDB
        	/usr/local/go/src/database/sql/sql.go:742 +0x12a
        
         Goroutine 5 in state select, with github.com/go-sql-driver/mysql.(*mysqlConn).startWatcher.func1 on top of the stack:
        goroutine 5 [select]:
        github.com/go-sql-driver/mysql.(*mysqlConn).startWatcher.func1(0xc0000261e0, 0xc0000d0a20, 0xc000022180)
        	/Users/ankur/go/pkg/mod/github.com/go-sql-driver/mysql@v1.5.0/connection.go:621 +0xbf
        created by github.com/go-sql-driver/mysql.(*mysqlConn).startWatcher
        	/Users/ankur/go/pkg/mod/github.com/go-sql-driver/mysql@v1.5.0/connection.go:618 +0xbe
        
         Goroutine 34 in state select, with github.com/sanketplus/go-mysql-lock.Lock.refresher on top of the stack:
        goroutine 34 [select]:
        github.com/sanketplus/go-mysql-lock.Lock.refresher(0x1372541, 0x3, 0xc000204060, 0xc000216000, 0x13eb440, 0xc0000e4a00, 0xc000099250, 0x3b9aca00, 0xc000099250)
        	/Users/ankur/workspaces/go-mysql-lock/lock.go:33 +0xb7
        created by github.com/sanketplus/go-mysql-lock.MysqlLocker.ObtainTimeoutContext
        	/Users/ankur/workspaces/go-mysql-lock/locker_client.go:100 +0x468
        ]
FAIL
FAIL	github.com/sanketplus/go-mysql-lock	0.915s
FAIL 
```

**Comment by sanketplus** [comment]
merged in #7
